syntax = "proto2";


message SetTemperature { 
    optional int32 new_temp = 1;
}

message SetTemperatureResult { 
    enum Result {
        OK = 0;
        Failed = 1;
    }
    optional Result result = 1;
} 

message ShowTemperatureResult { 
    enum Result {
        OK = 0;
        Failed = 1;
    }
    optional Result result = 1;
} 

message PowerOffResult { 
    enum Result {
        OK = 0;
        Failed = 1;
    }
    optional Result result = 1;
} 

message TurnOffDisplayResult { 
    enum Result {
        OK = 0;
        Failed = 1;
    }
    optional Result result = 1;
} 

message PowerOnResult { 
    enum Result {
        OK = 0;
        Failed = 1;
    }
    optional Result result = 1;
} 

message SystemCtrlMessagge {
    required int32 message_id = 1;
    oneof msg_body { 
        SetTemperature msg_set_temperature = 2; 
        SetTemperatureResult msg_set_temperature_result = 3; 
        ShowTemperatureResult msg_show_temperature_result = 4; 
        PowerOffResult msg_power_off_result = 5; 
        TurnOffDisplayResult msg_turn_off_display_result = 6; 
        PowerOnResult msg_power_on_result = 7;  
    }
}
